### PR TITLE
Speedup MSA with qualities calculation 

### PR DESCRIPTION
**Pull request by roger-liu-bina**
I got an almost 2x speedup. Heads up: there might be some small difference caused by different rounding. I remove some part like mapq first divided by 70.0 and then times 70.0. @msahraeian is that because you trying to avoid integer overflow? It should be ok if our depth is less than 10M. 
See the follow table. 

master neusomatic : 

With qualities
real    11m48.057s
user    11m13.059s
sys     0m31.734s

Without qualities
real    5m59.952s
user    5m20.002s
sys     0m38.527s

speedup DeepSomatic

With qualities
real    6m53.103s
user    6m12.258s
sys     0m39.265s

### CODE REVIEW

**Comment by msahraeian** [comment]
@roger-liu-bina the speed improvement sounds amazing. Thanks for quick hack. I will give it a try tomorrow. As you said, the goal of dividing to 70.0 and multiply to 70.0 was to avoiding overflow. Because we first sum them up and then take the average since we don't know the count at the beginning. I was trying to bring them between 0-1 and sum, then divide by count, then scale it back to 70.0. If you believe it is fine, you can remove it.

File: neusomatic/cpp/scan_alignments.cpp [hunk 1]
```diff
@@ -220,87 +223,89 @@ int main(int argc, char **argv) {
...
-          <<"\t"<<int(round(cols_tag[1][i].tag_mean['-']))<<":"<<int(round(cols_tag[1][i].tag_mean['A']))<<":"<<int(round(cols_tag[1][i].tag_mean['C'])) \
-          <<":"<<int(round(cols_tag[1][i].tag_mean['G']))<<":"<<int(round(cols_tag[1][i].tag_mean['T']))\
-          <<"\t"<<int(round(cols_tag[2][i].tag_mean['-']))<<":"<<int(round(cols_tag[2][i].tag_mean['A']))<<":"<<int(round(cols_tag[2][i].tag_mean['C'])) \
-          <<":"<<int(round(cols_tag[2][i].tag_mean['G']))<<":"<<int(round(cols_tag[2][i].tag_mean['T']))\
-          <<"\t"<<int(round(cols_tag[3][i].tag_mean['-']))<<":"<<int(round(cols_tag[3][i].tag_mean['A']))<<":"<<int(round(cols_tag[3][i].tag_mean['C'])) \
-          <<":"<<int(round(cols_tag[3][i].tag_mean['G']))<<":"<<int(round(cols_tag[3][i].tag_mean['T']))\
-          <<"\t"<<int(round(cols_tag[4][i].tag_mean['-']))<<":"<<int(round(cols_tag[4][i].tag_mean['A']))<<":"<<int(round(cols_tag[4][i].tag_mean['C'])) \
-          <<":"<<int(round(cols_tag[4][i].tag_mean['G']))<<":"<<int(round(cols_tag[4][i].tag_mean['T']))\
+          <<pileup_counts[4]<<":"<<pileup_counts[0]<<":"<<pileup_counts[1]<<":"<<pileup_counts[2] \
```
Comment (marghoob) [inline neusomatic/cpp/scan_alignments.cpp]: @msahraeian this is a giant cout with a large possibility of messing up. Should simplify this line. Something akin to [join](https://stackoverflow.com/questions/1833447/a-good-example-for-boostalgorithmjoin) would be useful. Can define a specific function for join.

Comment (msahraeian) [reply neusomatic/cpp/scan_alignments.cpp]: @marghoob is it good now?

File: neusomatic/cpp/scan_alignments.cpp [hunk 2]
```diff
@@ -220,87 +223,78 @@ int main(int argc, char **argv) {
...
-          <<"\t"<<int(round(cols_tag[1][i].tag_mean['-']))<<":"<<int(round(cols_tag[1][i].tag_mean['A']))<<":"<<int(round(cols_tag[1][i].tag_mean['C'])) \
-          <<":"<<int(round(cols_tag[1][i].tag_mean['G']))<<":"<<int(round(cols_tag[1][i].tag_mean['T']))\
-          <<"\t"<<int(round(cols_tag[2][i].tag_mean['-']))<<":"<<int(round(cols_tag[2][i].tag_mean['A']))<<":"<<int(round(cols_tag[2][i].tag_mean['C'])) \
-          <<":"<<int(round(cols_tag[2][i].tag_mean['G']))<<":"<<int(round(cols_tag[2][i].tag_mean['T']))\
-          <<"\t"<<int(round(cols_tag[3][i].tag_mean['-']))<<":"<<int(round(cols_tag[3][i].tag_mean['A']))<<":"<<int(round(cols_tag[3][i].tag_mean['C'])) \
-          <<":"<<int(round(cols_tag[3][i].tag_mean['G']))<<":"<<int(round(cols_tag[3][i].tag_mean['T']))\
-          <<"\t"<<int(round(cols_tag[4][i].tag_mean['-']))<<":"<<int(round(cols_tag[4][i].tag_mean['A']))<<":"<<int(round(cols_tag[4][i].tag_mean['C'])) \
-          <<":"<<int(round(cols_tag[4][i].tag_mean['G']))<<":"<<int(round(cols_tag[4][i].tag_mean['T']))\
-          <<std::endl;
+          << neusomatic::add_qual_col(pileup_counts, true)<<"\t" \
```
Comment (marghoob) [inline neusomatic/cpp/scan_alignments.cpp:234]: 👍

File: neusomatic/cpp/scan_alignments.cpp [hunk 3]
```diff
@@ -220,87 +223,78 @@ int main(int argc, char **argv) {
...
-        for(auto it = cols[i].base_freq_.cbegin(); it != cols[i].base_freq_.cend(); ++it) {
-          ntcounts.push_back(it->second);
+        int major = -1;
+        int major_count = 0;
+        int minor = -1;
+        int minor_count = 0;
+        int minor2 = -1;
+        int minor2_count = 0;
+
+        for (int row = 0;  row < cols[i].base_freq_.size(); ++row) {
```
Comment (marghoob) [inline neusomatic/cpp/scan_alignments.cpp:260]: @msahraeian what is this code segment doing?

**Comment by msahraeian** [comment]
@marghoob can we merge?

**Comment by marghoob** [comment]
@msahraeian yes

**Pull request merged**